Dockerfile

```diff
@@ -1,30 +1,41 @@
-# builder
-FROM golang:1.9.3-alpine3.7 AS builder
+# Accept the Go version for the image to be set as a build argument.
+# Default to Go 1.11
+ARG GO_VERSION=1.12
 
-WORKDIR /go/src/github.com/alicebob
+# First stage: build the executable.
+FROM golang:${GO_VERSION}-alpine AS build
 
-ARG ASPROM_VERSION=1.0.1
+# Create the user and group files that will be used in the running container to
+# run the process as an unprivileged user.
+RUN mkdir /user && \
+    echo '1000:x:65534:65534:1000:/:' > /user/passwd && \
+    echo '1000:x:65534:' > /user/group
 
-RUN apk add --no-cache git \
-    && git clone https://github.com/alicebob/asprom.git \
-    && cd asprom \
-    && git fetch --all --tags --prune \
-    && git checkout tags/$ASPROM_VERSION
+# Install the Certificate-Authority certificates for the app to be able to make
+# calls to HTTPS endpoints.
+RUN apk add --no-cache ca-certificates git curl
 
-WORKDIR /go/src/github.com/alicebob/asprom
+WORKDIR /go/src/github.com/alicebob
+
+ARG ASPROM_VERSION=1.0.1
 
-RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o asprom .
+COPY . .
 
+ENV GO111MODULE=on
```

Comment:
If you change the WORKDIR to be something like `/src/` this will use the go.mod stuff, and this setting should be not needed anymore.

Reply:
fixed